src/themes/default/components/core/blocks/Checkout/OrderReview.vue

```diff
@@ -61,7 +67,41 @@ export default {
   },
   methods: {
     placeOrder () {
-      this.$bus.$emit('checkout.placeOrder')
+      if (this.$store.state.checkout.personalDetails.createAccount) {
+        this.register()
+      } else {
+        this.$bus.$emit('checkout.placeOrder')
+      }
+    },
+    register () {
+      this.$bus.$emit('notification-progress-start', 'Registering the account ... ')
+      this.$store.dispatch('user/register', {
+        email: this.$store.state.checkout.personalDetails.emailAddress,
+        password: this.$store.state.checkout.personalDetails.password,
+        firstname: this.$store.state.checkout.personalDetails.firstName,
+        lastname: this.$store.state.checkout.personalDetails.lastName
+      }).then((result) => {
+        console.log(result)
+        this.$bus.$emit('notification-progress-stop')
+        if (result.code !== 200) {
+          this.$bus.$emit('notification', {
+            type: 'error',
+            message: result.result,
+            action1: { label: 'OK', action: 'close' }
+          })
+        } else {
+          this.$bus.$emit('notification', {
+            type: 'success',
+            message: 'You are logged in!',
+            action1: { label: 'OK', action: 'close' }
+          })
```

Comment:
We should add the user.id to the order object - it's not implemented yet to pass the non-guest orders to the Magento backend yet; i belive we should add next section to the order object like `user` with `id` property. The whole section shouldn't be required - please modify the models/order.schema.json accordingly (and also update the `vue-storefront-api` copy of the schema)